tests/unittest/_torch/visual_gen/multi_gpu/test_visual_gen_mapping.py

```
@@ -507,6 +524,50 @@ def _logic_attn2d_seq_rank_matches_global_rank(rank, world_size):
     )
 
 
+def _logic_tp2_attn2d_2x1_groups(rank, world_size):
+    """Groups compose under tp=2 × attn2d(2×1) on 4 GPUs (guard removed).
```

Ruff (0.15.21)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)